gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/CopyJobInfo.java

```diff
@@ -0,0 +1,255 @@
+    return sourceTables;
+  }
+
+  /**
+   * Returns the destination table to load the data into.
+   */
+  public TableId destinationTable() {
+    return destinationTable;
+  }
+
+  /**
+   * Returns whether the job is allowed to create new tables.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.copy">
+   *     Jobs: Copy Configuration</a>
+   */
+  public CreateDisposition createDisposition() {
+    return this.createDisposition;
+  }
+
+  /**
+   * Returns the action that should occur if the destination table already exists.
+   *
+   * @see <a href="https://cloud.google.com/bigquery/docs/reference/v2/jobs#configuration.copy">
+   *     Jobs: Copy Configuration</a>
+   */
+  public WriteDisposition writeDisposition() {
+    return writeDisposition;
+  }
+
+  @Override
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  @Override
+  public MoreObjects.ToStringHelper toStringHelper() {
+    return super.toStringHelper()
+        .add("sourceTables", sourceTables)
+        .add("destinationTable", destinationTable)
+        .add("createDisposition", createDisposition)
+        .add("writeDisposition", writeDisposition);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(super.hashCode(), sourceTables, destinationTable, createDisposition,
+        writeDisposition);
+  }
+
+  @Override
+  Job toPb() {
+    Job jobPb = super.toPb();
+    JobConfiguration configurationPb = jobPb.getConfiguration();
+    JobConfigurationTableCopy copyConfigurationPb = new JobConfigurationTableCopy();
+    copyConfigurationPb.setDestinationTable(destinationTable.toPb());
+    if (sourceTables.size() == 1) {
+      copyConfigurationPb.setSourceTable(sourceTables.get(0).toPb());
+    } else {
+      copyConfigurationPb.setSourceTables(Lists.transform(sourceTables, TableId.TO_PB_FUNCTION));
+    }
+    if (createDisposition != null) {
+      copyConfigurationPb.setCreateDisposition(createDisposition.toString());
+    }
+    if (writeDisposition != null) {
+      copyConfigurationPb.setWriteDisposition(writeDisposition.toString());
+    }
+    return jobPb.setConfiguration(configurationPb.setCopy(copyConfigurationPb));
+  }
+
+  /**
+   * Creates a builder for a BigQuery Copy Job given destination and source table.
+   */
+  public static Builder builder(TableId destinationTable, TableId sourceTable) {
+    return builder(destinationTable, ImmutableList.of(checkNotNull(sourceTable)));
+  }
+
+  /**
+   * Creates a builder for a BigQuery Copy Job given destination and source tables.
+   */
+  public static Builder builder(TableId destinationTable, List<TableId> sourceTables) {
+    return new Builder().destinationTable(destinationTable).sourceTables(sourceTables);
+  }
+
+  /**
+   * Returns a BigQuery Copy Job for the given destination and source table. Job's id is chosen by
+   * the service.
+   */
+  public static CopyJobInfo of(TableId destinationTable, TableId sourceTable) {
```

Comment:
In other modules, we test the `of` functions (even though they're pretty simple), [for example in storage](https://github.com/GoogleCloudPlatform/gcloud-java/blob/a25e50430b55b93816be392e363638e67f67a500/gcloud-java-storage/src/test/java/com/google/gcloud/storage/BucketInfoTest.java#L98).  I think it'd be good to add tests for the `of` functions here (and in the other classes) as well.